lib/web/resources.go

```diff
@@ -306,7 +306,7 @@ func CheckResourceUpsertableByError(err error, httpMethod, resourceName string)
 	}
 
 	if !exists && httpMethod == http.MethodPut {
-		return trace.NotFound("cannot find resource with name %q", resourceName)
+		return trace.NotFound("resource renaming is not supported, please create a different resource and then delete this one")
```

Comment:
Is name the only attribute that can be changed by `http.MethodPut`? `Cannot find resource` seems to fit better as an error message if checking its existence returns false, but if the only thing we can update is name anyway, then this is better.

Reply:
`http.MethodPut` is used for editing the resource. All attributes can be changed (I believe), but not the name, so renaming is not supported.

The current error message for when the user tries to rename the resource is not super friendly, as detailed in #18771. Since we don't support renaming, this PR improves the error message indicating that.

Reply:
Is there a better place to detect a rename rather than checking if the resource exists?

Technically the resource has an ID field, so if the ID has stayed the same but the name changed that might be a better way to detect this.

Reply:
Server-side, this seems to be the best way to do it (IIUC there's no way to find a resource given its id).
However, the validation you suggested could be done on the client.

Reply:
Client-side might be the best place for this validation actually.

If we keep the validation as is on the server we can have the following situation.

Given two roles with names `roleA` and `roleB`, if the user renames `roleA` to `roleB`, then:
- the contents of `roleB` will be overwritten with the contents of `roleA`
- `roleA` will still exist as before